GameOfLife/GameOfLifeUI/Program.cs

```diff
@@ -0,0 +1,63 @@
+﻿using GameOfLifeProcessor;
+using GameOfLifeProcessor.GameBoard;
+using GameOfLifeUI;
+using Pastel;
+using System.Drawing;
+
+//Settings
+var generations = 150;
+var boardWidth = 25;
+var boardHeight = 10;
+var delayOutput = true;
+var delayInMs = 100;
+var cellCharacter = "x";
+var clearOutput = true;
+var cellTextColor = Color.FromArgb(0, 0, 0);
+var activeCellTextColor = Color.FromArgb(66, 245, 72);
+var randomBoard = true;
+var activeCells = ActiveCellsExamples.Beacon;
+
+//Game
+try
+{
+    var game = randomBoard ? new Game(boardWidth, boardHeight) : new Game(boardWidth, boardHeight, activeCells);
+    for (var i = 0; i < generations; i++)
+    {
+        Print(game.GetBoard());
+        game.NextGeneration();
+
+        if (delayOutput)
+        {
+            Thread.Sleep(delayInMs);
+        }
+
+        if (clearOutput && i + 1 < generations)
+        {
+            Console.Clear();
+        }
+    }
+}
+catch(Exception ex)
+{
+    Console.WriteLine($"An error occurred: {ex.Message}".Pastel(Color.FromArgb(247, 52, 52)));
+}
+
+//Output method
+void Print(Board board)
```

Comment:
Ich weiß, das hier ist eben die Program.cs,
aber wenn du die Aufgaben klar trennen willst, braucht auch die Program.cs in dem UI Projekt,
nicht mehr als den Startcode, Parameter auslesen, weitergeben, starten.

Konkret verstößt diese Klasse hier gegen das S (Single Responsibility),
weil sie:
- Daten definiert
- Daten initialisiert
- den Game Loop vornimmt
- Daten ausgibt

Reply:
aber das ist schon meckern auf hohem niveau,
stell dir vor, du machst ein zweites UI Projekt, mit einer anderen Technologie, z.B. WPF,
und weißt nicht wie das GameOfLife funktioniert, du bekommst nur den GameOfLifeProcessor.
könntest du damit aktuell ohne groß Erklärung dies implementieren?

Ich weiß, abstraktion ist schwierig und teilweise auch echt von der technologie abhängig,
wenn zu stark abstrahiert (Controlling, Logik, Datalayer), kannst es vllt nur begrenzt wiederverwenden, wenn zu leicht, implementierst du dinge mehrfach, bist aber flexibler (nur Datalayer/grobe Logikschicht).